test/framework/CMakeLists.txt

```diff
@@ -24,12 +24,18 @@ target_link_libraries(test_subscriber_testing
 add_library(integration_framework
     integration_framework/integration_test_framework.cpp
     integration_framework/iroha_instance.cpp
+    integration_framework/fake_peer/fake_peer.cpp
```

Comment:
I think that would only increase the compilation time, because the linkage would be done against several identical libs. also the fake peer is not intended to be used without the ITF, and the ITF now cannot be built without fake peer. So I think these two are better to be built this way.